src/mlpack/core/math/lin_alg.cpp

```diff
@@ -257,7 +257,9 @@ void mlpack::math::Svec(const arma::sp_mat& input, arma::sp_vec& output)
 
 void mlpack::math::Smat(const arma::vec& input, arma::mat& output)
 {
-  const size_t n = static_cast<size_t>(ceil((-1. + sqrt(1. + 8. * input.n_elem))/2.));
+  const size_t n = static_cast<size_t>
```

Comment:
I think that without the `static_cast` a compiler warning may be generated, but if it doesn't I agree that we can put this all on one line.  If it does, I think the opening `(` of the cast can be on the first line.